osu.Game/Overlays/Rankings/Tables/CountriesTable.cs

```diff
@@ -62,35 +61,19 @@ public CountriesTable(int page, IReadOnlyList<CountryStatistics> rankings)
             }
         };
 
-        private class CountryName : OsuHoverContainer
+        private class CountryName : LinkFlowContainer
         {
-            protected override IEnumerable<Drawable> EffectTargets => new[] { text };
-
             [Resolved(canBeNull: true)]
             private RankingsOverlay rankings { get; set; }
 
-            private readonly OsuSpriteText text;
-            private readonly Country country;
-
             public CountryName(Country country)
+                : base(t => t.Font = OsuFont.GetFont(size: 12))
             {
-                this.country = country;
-
-                AutoSizeAxes = Axes.Both;
-                Add(text = new OsuSpriteText
-                {
-                    Font = OsuFont.GetFont(size: 12),
-                    Text = country.FullName ?? string.Empty,
-                });
-            }
-
-            [BackgroundDependencyLoader]
-            private void load(OverlayColourProvider colourProvider)
-            {
-                IdleColour = colourProvider.Light2;
-                HoverColour = colourProvider.Content2;
+                AutoSizeAxes = Axes.X;
+                RelativeSizeAxes = Axes.Y;
+                TextAnchor = Anchor.CentreLeft;
 
-                Action = () => rankings?.ShowCountry(country);
+                AddLink(country.FullName ?? string.Empty, () => rankings?.ShowCountry(country));
```

Comment:
Now that I look at this, wouldn't it be better if it only added a link when the `country.FullName` has a value? I kinda mindlessly copied the text value from the previous version and didn't notice that until now, I'm also not sure if it can ever realistically be null (haven't checked).

Reply:
It would.